WordPress/Classes/ViewRelated/Comments/CommentViewController.m

```diff
@@ -504,30 +513,23 @@ - (void)trashComment
     [self presentViewController:alertController animated:YES completion:nil];
 }
 
-- (void) deleteAction
+- (void)deleteAction
 {
     __typeof(self) __weak weakSelf = self;
     NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
     CommentService *commentService = [[CommentService alloc] initWithManagedObjectContext:context];
     
-    NSError *error = nil;
-    Comment *reloadedComment = (Comment *)[context existingObjectWithID:self.comment.objectID error:&error];
-    
-    if (error) {
-        DDLogError(@"Comment was deleted while awaiting for alertView confirmation");
-        return;
-    }
-    
     [CommentAnalytics trackCommentTrashedWithComment:[self comment]];
     
-    [commentService deleteComment:reloadedComment success:^{
+    [commentService deleteComment:self.comment success:^{
```

Comment:
Huh. I wonder why we were doing this.

Reply:
I finally figured out (after this PR) the weirdness I was seeing is because if you Trash a Spam comment, the comment is _permanently_ deleted. Our alerts don't clarify that (which I'll be fixing). So the Spam and Trash actions were reloading the comment to make sure it existed. But the other actions weren't. And even if they did, because we continued to display the comment details, it appeared to still existed and thus weirdness.